javascript/ql/src/semmle/javascript/dataflow/Nodes.qll

```diff
@@ -975,3 +974,134 @@ module ClassNode {
     }
   }
 }
+
+/**
+ * A data flow node that performs a partial function application.
+ *
+ * Examples:
+ * ```js
+ * fn.bind(this)
+ * x.method.bind(x)
+ * _.partial(fn, x, y, z)
+ * ```
+ */
+class PartialInvokeNode extends DataFlow::Node {
+  PartialInvokeNode::Range range;
+
+  PartialInvokeNode() { this = range }
+
+  /**
+   * Holds if `argument` is passed as argument `index` to the function in `callback`.
+   */
+  predicate isPartialArgument(DataFlow::Node callback, DataFlow::Node argument, int index) {
+    range.isPartialArgument(callback, argument, index)
+  }
+
+  /**
+   * Gets a node referring to a bound version of `callback` with `boundArgs` arguments bound.
+   */
+  DataFlow::SourceNode getBoundFunction(DataFlow::Node callback, int boundArgs) {
+    result = range.getBoundFunction(callback, boundArgs)
+  }
+
+  /**
+   * Gets the node holding the receiver to be passed to the bound function, if specified.
+   */
+  DataFlow::Node getBoundReceiver() { result = range.getBoundReceiver() }
+}
+
+module PartialInvokeNode {
+  /**
+   * A data flow node that performs a partial function application.
```

Comment:
```suggestion
   * A data-flow node that performs a partial function application.
```

Reply:
Ditto in a few more places below.

Reply:
There are 238 occurrences of `data flow node`. I'll do it in a separate PR to avoid conflicts.